### ISSUE TITLE
Inserts trailing comma into TypeScript Type Parameters

### ISSUE THREAD

**Issue by azz** [prettier/prettier#6189]
Due to #6115, after upgrading to 1.18, Prettier does this in `.tsx` files:

```diff
-export interface Group<T> {
+export interface Group<T,> {
  label: string;
  options: T[];
}
```

**Comment by duailibe** [comment]
@azz Should we revert it?

**Comment by duailibe** [comment]
Oh I understand now.. we need to make it more restricted

**Comment by azz** [comment]
Yeah, not sure what the rule/heuristic should be.

**Comment by duailibe** [comment]
cc @sosukesuzuki

**Comment by azz** [comment]
Maybe only need to add the comma for function type arguments? `<T,>()`

**Comment by gfx** [comment]
I think it should behave just like other trailing commas. That is, add trailing commas only before `\n`.

**Comment by azz** [comment]
It already does, this is a workaround for a specific bug (#6114)

**Comment by sosukesuzuki** [comment]
I see. Sorry, this is my mistake. I think we should add comma for only function type arguments.

**Comment by duailibe** [comment]
Fixed in 1.18.1

**Comment by kachkaev** [comment]
Thanks @duailibe. Funny enough, I just opened a PR in an internal project, which updates prettier to 1.18.0. While I was waiting for CI to complete, it was time to bump Prettier again 😅
